src/main/java/de/unistuttgart/informatik/fius/icge/simulation/Mario.java

```diff
@@ -8,22 +8,122 @@
 package de.unistuttgart.informatik.fius.icge.simulation;
 
 import de.unistuttgart.informatik.fius.icge.simulation.inspection.InspectionAttribute;
+import de.unistuttgart.informatik.fius.icge.simulation.inspection.InspectionMethod;
 
-
+/**
+ * A mario entity
+ * 
+ * @author haslersn, neumantm
+ */
 public class Mario extends GreedyEntity {
-    
+
+    /**
+     * Creates a new mario with in given simulation
+     * 
+     * @param sim
+     *            The simulation for the new mario.
+     */
     public Mario(Simulation sim) {
         super(sim, EntityType.MARIO);
     }
-    
+
+    /**
+     * Creates a new mario with the given initial coin count in the given simulation.
+     * 
+     * @param sim
+     *            The simulation for the new mario.
+     * @param coinCount
+     *            The initial count of coins of mario.
+     */
+    public Mario(Simulation sim, int coinCount) {
+        super(sim, EntityType.MARIO);
+        setCoinCount(coinCount);
+    }
+
     @Override
     public boolean canCollectType(EntityType type) {
         return type == EntityType.COIN;
     }
-    
+
     @Override
     public boolean canDropType(EntityType type) {
         return type == EntityType.COIN;
     }
 
+    /**
+     * Get the coin count of mario.
+     * 
+     * @return The coin count
+     */
+    @InspectionAttribute
+    public int getCoinCount() {
+        return this._inventory.size();
```

Comment:
No because mario can only collect coins. See https://github.com/FIUS/ICGE/pull/36/files#diff-4c6e14173992784be1f68f93ba6cd990R45
If this is changed it is obvious to also change the getCoinCount()